File: app/src/main/java/dev/ragnarok/fenrir/util/AssertUtils.java
package dev.ragnarok.fenrir.util;

public class AssertUtils {

    /**
     * Returns {@code o} if non-null, or throws {@code NullPointerException}.
     */
    public static <T> T requireNonNull(T o) {
        if (o == null) {
            throw new NullPointerException();
        }
        return o;
    }

    /**
     * Returns {@code o} if non-null, or throws {@code NullPointerException}
     * with the given detail message.
     */
    public static <T> T requireNonNull(T o, String message) {
        if (o == null) {
            throw new NullPointerException(message);
        }
        return o;
    }

    public static void assertPositive(int value) {
        if (value <= 0) {
            throw new IllegalStateException();
        }
    }

    public static void assertTrue(boolean value) {
        if (!value) {
            throw new IllegalStateException();
        }
    }
}

File: app_filegallery/src/main/kotlin/dev/ragnarok/filegallery/util/AssertUtils.kt
package dev.ragnarok.filegallery.util

object AssertUtils {
    /**
     * Returns `o` if non-null, or throws `NullPointerException`.
     */
    fun <T> requireNonNull(o: T?): T {
        if (o == null) {
            throw NullPointerException()
        }
        return o
    }

    /**
     * Returns `o` if non-null, or throws `NullPointerException`
     * with the given detail message.
     */
    fun <T> requireNonNull(o: T?, message: String?): T {
        if (o == null) {
            throw NullPointerException(message)
        }
        return o
    }

    fun assertPositive(value: Int) {
        check(value > 0)
    }

    fun assertTrue(value: Boolean) {
        check(value)
    }
}